client/client-lib/src/mint/mod.rs

```diff
@@ -242,7 +248,7 @@ impl MintClient {
     ) where
         F: FnMut(TieredMulti<BlindNonce>) -> Fut,
         Fut: futures::Future<Output = OutPoint>,
-        G: FnMut() -> GFut,
+        G: FnMut(Amount) -> GFut,
```

Comment:
Sorry you had to suffer through extending this mess. Would I have known we'll touch it that often before completely refactoring the client I'd have done it more properly.

Reply:
It went pretty quickl. The only problem here is that Rust doesn't have async closures yet, so `|amount| async { ... }` actually borrows `amount` which is extra weird, and if you change to `|amount| async move { ... }` then it will move whatever is used inside `{...}` and you have to explicitly borrow it outside. :shrug: